### PR TITLE
docs: add ORAS implementation

### PR DESCRIPTION
**Pull request by Wwwsylvia**
Adding the existing auth implementation of [ORAS ](https://github.com/oras-project/oras)(backed by [`oras-go v2`](https://github.com/oras-project/oras-go)).
Related issue: https://github.com/oras-project/oras-credentials-go/issues/97

### CODE REVIEW

**Comment by sudo-bmitch** [comment]
This is looking good. The linter picked up a few small nits. When you're ready for a review, please squash the commits.

File: docs/implementations/ORAS.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+- Merge the challenged `<scope>` with the built-in known scopes to get `<result-scopes>`
+  - For example, the challenged is `"repository:<repo>:pull,push"` and the built-in known scopes is `"repository:<repo>:delete,pull"`, the merged scopes will be `"repository:<repo>:delete,pull,push"`
+- If `<result-scopes>` is different than the built-in known scopes:
+  - Try to load token again for `<result-scopes>` from cache
+  - If cache hit:
+    - Set the `Authorization` header to `"Bearer <cached-token>"`
+    - Request `/v2/<api>` with the `Authorization` header
+- If `<result-scopes>` is the same as the built-in known scopes, or cache missed:
+  - Read`<realm>` `<username>`, `<password>`, `<identity-token>` and `<registry-token>` from user input or configured credentials stores
```
Comment (shizhMSFT) [inline docs/implementations/ORAS.md:42]: nit: 
```suggestion
  - Read `<realm>` `<username>`, `<password>`, `<identity-token>` and `<registry-token>` from user input or configured credentials stores
```

**Review by shizhMSFT — APPROVED** [review]
LGTM but IANAM

File: docs/implementations/ORAS.md (hunk 1 above)
Comment (Wwwsylvia) [reply docs/implementations/ORAS.md]: Nice catch! Fixed!

**Comment by Wwwsylvia** [comment]
Thanks @sudo-bmitch and @shizhMSFT . I've fixed the lint errors and squash the commits. This PR is ready for review now.

**Review by shizhMSFT — APPROVED** [review]
LGTM but IANAM

**Comment by sajayantony** [comment]
I would consider merging this and maybe expanding with 
1. clarify the path used by mount. 
2. Does oras use form fields? 

Again thanks @Wwwsylvia.

**Review by sudo-bmitch — APPROVED** [review]
LGTM

**Comment by Wwwsylvia** [comment]
@sajayantony ,

> 1. clarify the path used by mount. 

Since ORAS currently does not support repo-level auth, the auth path for mount is exactly the same as the regular path.

> 2. Does oras use form fields?

Could you help to elaborate this a bit more? What does "form fields" refer to?

**Comment by sudo-bmitch** [comment]
> 2. Does oras use form fields? 

I believe the answer is yes, based on the "Fetch Oauth2 Tokens" section.

**Review by jcarter3 — APPROVED** [review]

**Pull request merged**